add inode into the file fingerprint

We recently had to deal with improving decision making on either to reload config in datalad where in effect we also rely on similar concept.  We ended up with adding `st_ino` which helped greatly at least in case of `git config` which seems to just write a new file (produce a new inode) on modifications of `.git/config`, which otherwise we failed to detect:  https://github.com/datalad/datalad/commit/0ead554d91559ffdae9a312067e421ca5b71bfc0#diff-94f9b0e4cdbcfe44b244bbf4450719d1034680f27d3a1cef81393694ed7d6e08R378

I even wonder (blindly, didn't analyze) if that would somehow relate to test failures in the mothership dandi-cli like this one from https://github.com/dandi/dandi-cli/pull/330/checks?check_run_id=1723505920
```
dandi/support/tests/test_cache.py:121: in test_memoize_path
    check_new_memoread(0, "content", True)
dandi/support/tests/test_cache.py:101: in check_new_memoread
    assert len(calls) == ncalls + 1 + int(expect_new)
E   assert 3 == 4
E     +3
E     -4
```

--- a/src/fscacher/cache.py
+++ b/src/fscacher/cache.py
@@ -155,10 +155,10 @@ class PersistentCache(object):
             return fprint
 
 
-class FileFingerprint(namedtuple("FileFingerprint", "mtime_ns ctime_ns size")):
+class FileFingerprint(namedtuple("FileFingerprint", "mtime_ns ctime_ns size inode")):
     @classmethod
     def from_stat(cls, s):
-        return cls(s.st_mtime_ns, s.st_ctime_ns, s.st_size)
+        return cls(s.st_mtime_ns, s.st_ctime_ns, s.st_size, s.st_ino)
 
     def modified_in_window(self, min_dtime):
         return abs(time.time() - self.mtime_ns * 1e-9) < min_dtime

